src/earn/prepareTransactions.ts

```diff
@@ -134,6 +138,24 @@ export async function prepareSupplyTransactions({
   })
 }
 
+/**
+ * Hook to prepare transactions for supplying crypto.
+ */
+export function usePrepareSupplyTransactions() {
+  const prepareTransactions = useAsyncCallback(prepareSupplyTransactions, {
+    onError: (error) => {
+      Logger.error(TAG, `usePrepareSupplyTransactions: ${error}`)
```

Comment:
```suggestion
      Logger.error(TAG, 'usePrepareSupplyTransactions', error)
```
so the error is logged with trace, you may need an ensureError potentially